daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/parsers/Parser.scala

```diff
@@ -90,6 +85,53 @@ sealed trait Parser extends Processor {
     )
   }
 
+  def PENotEnoughBits(
```

Comment:
I wonder if we should remove the above `PENotEnoughBits` that accepts a `remainingBits` parameer? Nothing should every calculate their own remaining bits and call that since they'll likely do it wrong and not use information from DataInputStream, DataLocation, etc.

Reply:
There is one place that passes in a None or Nope here so I created a new constructor that doesn't take this argument in that case.